hedera-mirror-rest/controllers/contractController.js

```diff
@@ -446,426 +424,491 @@ const extractContractIdAndFiltersFromValidatedRequest = (req) => {
- * @returns {Promise<void>}
- */
-const getContractResultsByTimestamp = async (req, res) => {
-  const {timestamp} = getAndValidateContractIdAndConsensusTimestampPathParams(req);
-
-  // retrieve contract result, recordFile and transaction models concurrently
-  const [contractResults, recordFile, transaction, contractLogs, contractStateChanges] = await Promise.all([
-    ContractService.getContractResultsByTimestamps(timestamp),
-    RecordFileService.getRecordFileBlockDetailsFromTimestamp(timestamp),
-    TransactionService.getTransactionDetailsFromTimestamp(timestamp),
-    ContractService.getContractLogsByTimestamps(timestamp),
-    ContractService.getContractStateChangesByTimestamps(timestamp),
-  ]);
-  if (_.isNil(transaction)) {
-    throw new NotFoundError('No correlating transaction');
-  }
+    res.locals[constants.responseDataLabel] = response;
+  };
 
-  if (contractResults.length === 0) {
-    throw new NotFoundError();
-  }
+  /**
+   * Handler function for /contracts/:contractId/results/logs API
+   * @param {Request} req HTTP request object
+   * @param {Response} res HTTP response object
+   * @returns {Promise<void>}
+   */
+  getContractLogsById = async (req, res) => {
+    // get sql filter query, params, limit and limit query from query filters
+    const {filters, contractId: contractIdParam} = extractContractIdAndFiltersFromValidatedRequest(req);
+    checkTimestampsForTopics(filters);
+
+    const contractId = await ContractService.computeContractIdFromString(contractIdParam);
+
+    const {conditions, params, timestampOrder, indexOrder, limit} = this.extractContractLogsQuery(filters, contractId);
+
+    const rows = await ContractService.getContractLogsByIdAndFilters(
+      conditions,
+      params,
+      timestampOrder,
+      indexOrder,
+      limit
+    );
 
-  if (_.isNil(contractResults[0].callResult)) {
-    // set 206 partial response
-    res.locals.statusCode = httpStatusCodes.PARTIAL_CONTENT.code;
-    logger.debug(`getContractResultsByTimestamp returning partial content`);
-  }
+    res.locals[constants.responseDataLabel] = {
+      logs: rows.map((row) => new ContractLogViewModel(row)),
+    };
+  };
 
-  res.locals[constants.responseDataLabel] = new ContractResultDetailsViewModel(
-    contractResults[0],
-    recordFile,
-    transaction,
-    contractLogs,
-    contractStateChanges
-  );
-};
+  /**
+   * Handler function for /contracts/results/logs API
+   * @param {Request} req HTTP request object
+   * @param {Response} res HTTP response object
+   * @returns {Promise<void>}
+   */
+  getContractLogs = async (req, res) => {
+    // get sql filter query, params, limit and limit query from query filters
+    const filters = this.extractFiltersFromValidatedRequest(req);
+    checkTimestampsForTopics(filters);
+
+    const {conditions, params, timestampOrder, indexOrder, limit} = this.extractContractLogsQuery(filters);
+
+    const rows = await ContractService.getContractLogsByIdAndFilters(
```

Comment:
nit, remove `Id` from the function name,
```suggestion
    const rows = await ContractService.getContractLogs(
```